contracts/ERC20SimpleSwap.sol

```diff
@@ -62,7 +56,7 @@ contract ERC20SimpleSwap {
   );
 
   // the EIP712 domain this contract uses
-  function domain() internal pure returns (EIP712Domain memory) {
+  function domain() internal view returns (EIP712Domain memory) {
     uint256 chainId;
     assembly {
```

Comment:
Solidity 0.8 exposes `block.chainid`, see: https://docs.soliditylang.org/en/v0.8.0/units-and-global-variables.html

Reply:
nice